src/client/dfuse/dfuse_cont.c

```diff
@@ -48,80 +50,100 @@ dfuse_cont_open(fuse_req_t req, struct dfuse_inode_entry *parent,
 		D_GOTO(err, rc = ENOMEM);
 	}
 
+	dfs->dfs_dfp = dfp;
+
 	/* Dentry names where are not valid uuids cannot possibly be added so in
 	 * this case return the negative dentry with a timeout to prevent future
 	 * lookups.
 	 */
 	if (uuid_parse(name, dfs->dfs_cont) < 0) {
 		struct fuse_entry_param entry = {.entry_timeout = 60};
 
-		DFUSE_LOG_ERROR("Invalid container uuid");
+		DFUSE_TRA_INFO(parent, "Invalid container uuid");
 		DFUSE_REPLY_ENTRY(req, entry);
 		D_FREE(dfs);
 		return;
 	}
-	uuid_copy(dfs->dfs_pool, parent->ie_dfs->dfs_pool);
 
 	DFUSE_TRA_UP(dfs, fs_handle, "dfs");
 
+	D_MUTEX_LOCK(&fs_handle->dpi_info->di_lock);
+
 	if (create) {
-		rc = dfs_cont_create(parent->ie_dfs->dfs_poh, dfs->dfs_cont,
+		rc = dfs_cont_create(dfp->dfp_poh, dfs->dfs_cont,
 				     NULL, NULL, NULL);
 		if (rc) {
-			DFUSE_LOG_ERROR("dfs_cont_create() failed: (%d)", rc);
-			D_GOTO(err, rc);
+			DFUSE_TRA_ERROR(dfs, "dfs_cont_create() failed: (%d)", rc);
+			D_GOTO(err_unlock, rc);
 		}
 	} else {
-		rc = dfuse_check_for_inode(fs_handle, dfs, &ie);
-		if (rc == -DER_SUCCESS) {
-			struct fuse_entry_param	entry = {0};
-
-			DFUSE_TRA_INFO(ie, "Reusing existing container entry "
-				       "without reconnect");
-
-			/* Update the stat information, but copy in the
-			 * inode value afterwards.
-			 */
-			rc = dfs_ostat(ie->ie_dfs->dfs_ns,
-				       ie->ie_obj, &entry.attr);
-			if (rc) {
-				DFUSE_TRA_ERROR(ie, "dfs_ostat() failed: (%s)",
-						strerror(rc));
-				d_hash_rec_decref(&fs_handle->dpi_iet,
-						  &ie->ie_htl);
-				D_GOTO(err, rc);
-			}
 
-			entry.attr.st_ino = ie->ie_stat.st_ino;
-			entry.generation = 1;
-			entry.ino = entry.attr.st_ino;
-			DFUSE_REPLY_ENTRY(req, entry);
-			D_FREE(dfs);
-			return;
+		d_list_for_each_entry(dfsi,
+				      &dfp->dfp_dfs_list,
+				      dfs_cont_list) {
+			{
+				struct fuse_entry_param	entry = {0};
+
+				DFUSE_TRA_DEBUG(parent, "Checking %p", dfsi);
+
+				if (uuid_compare(dfsi->dfs_cont, dfs->dfs_cont) != 0 )
```

Comment:
(style)  line over 80 characters
(style)  space prohibited before that close parenthesis ')'